mkt/langpacks/tests/test_views.py

```diff
@@ -0,0 +1,463 @@
+
+    def setUp(self):
+        super(TestLangPackViewSetMixin, self).setUp()
+        self.list_url = reverse('api-v2:langpack-list')
+        self.user = UserProfile.objects.get(pk=2519)
+
+    def create_langpack(self, **kwargs):
+        data = {
+            'filename': 'dummy.zip',
+            'hash': 'dummy-hash',
+            'size': 666,
+            'active': True,
+            'version': '0.1',
+            'language': 'fr',
+            'fxos_version': '2.2',
+        }
+        data.update(kwargs)
+        return LangPack.objects.create(**data)
+
+    def check_langpack(self, langpack_data, instance=None):
+        if instance is None:
+            instance = self.langpack
+        eq_(instance.pk, langpack_data['uuid'])
+        # FIXME: To implement in bug 1122279, don't expose the filename, expose
+        # the minifest_url instead.
+        eq_(instance.filename, langpack_data['filename'])
+        eq_(instance.hash, langpack_data['hash'])
+        eq_(instance.size, langpack_data['size'])
+        eq_(instance.active, langpack_data['active'])
+        eq_(instance.language, langpack_data['language'])
+        eq_(instance.fxos_version, langpack_data['fxos_version'])
+
+
+class TestLangPackViewSetBase(TestLangPackViewSetMixin):
+    def setUp(self):
+        super(TestLangPackViewSetBase, self).setUp()
+        self.detail_url = reverse('api-v2:langpack-detail', kwargs={'pk': 42})
+
+    def test_cors(self):
+        self.assertCORS(self.anon.options(self.detail_url),
+                        'get', 'delete', 'patch', 'post', 'put')
+        self.assertCORS(self.anon.options(self.list_url),
+                        'get', 'delete', 'patch', 'post', 'put')
+
+    def test_no_double_slash(self):
+        ok_(not self.detail_url.endswith('//'))
+        ok_(not self.list_url.endswith('//'))
+
+
+class TestLangPackViewSetGet(TestLangPackViewSetMixin):
+    fixtures = fixture('user_2519')
+
+    def setUp(self):
+        super(TestLangPackViewSetGet, self).setUp()
+        self.langpack = self.create_langpack()
+        self.detail_url = reverse('api-v2:langpack-detail',
+                                  kwargs={'pk': self.langpack.pk})
+
+    # Anonymously, you can view all active langpacks.
+    # Logged in view the right permission ('LangPacks', '%') you get them
+    # all if you use active=0.
+
+    def test_list_active(self):
+        response = self.anon.get(self.list_url)
+        eq_(response.status_code, 200)
+        ok_(len(response.json['objects']), 1)
+        self.check_langpack(response.json['objects'][0])
+
+        response = self.client.get(self.list_url)
+        eq_(response.status_code, 200)
+        ok_(len(response.json['objects']), 1)
+        self.check_langpack(response.json['objects'][0])
+
+        response = self.client.get(self.list_url)
+        eq_(response.status_code, 200)
+        ok_(len(response.json['objects']), 1)
+        self.check_langpack(response.json['objects'][0])
```

Comment:
This block looks the same as the block above it. Curious if you intended the detail url there?